File: admin/src/app/models/create-requests/product-detail-create-request.ts
import { AutoMap } from "@automapper/classes";
import { BaseCreateRequest } from "..";

export class ProductDetailCreateRequest extends BaseCreateRequest {
    @AutoMap()
    productId: number ;
    @AutoMap()
    langId: string ;
    @AutoMap()
    name: string ;
    @AutoMap()
    description: string ;
}

File: App.Models/DTOs/CreateRequests/ProductDetailCreateRequest.cs
using MaxV.Base.DTOs;
using Newtonsoft.Json;

namespace App.Models.DTOs.CreateRequests
{
    public class ProductDetailCreateRequest : BaseCreateRequest
    {
        [JsonProperty(PropertyName = "langId", Required = Required.Always)]
        public string LangId { get; set; }
        [JsonRequired]
        public string Name { get; set; }
        [JsonRequired]
        public string Description { get; set; }
    }
}
